### PR TITLE
Added library functions and test for certificate details verification

### PR DESCRIPTION
**Pull request by aditipande96**
Signed-off-by: Aditi Pande <aditi.pande@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by aditipande96** [comment]
Collection logs:
========= 2447 tests collected in 11.46s ====

Test log:
FAILED tests/csm/rest/test_query_deployment.py::TestQueryDeployment::test_45674 - AssertionError: Serial number match failed(possible defect)

PASSED tests/csm/rest/test_query_deployment.py::TestQueryDeployment::test_45674(without serial number check)

File: libs/csm/rest/csm_rest_query_deployment.py [hunk 1]
```diff
@@ -207,3 +223,196 @@ def verify_storage_set(self, storage_set_id: str = None,
             result = False
         return resp, result, err_msg
 
+    def verify_datetime_details(self, validity_str: str, get_response_date,
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_query_deployment.py:226]: Can u add data type for all parameter
eta whereever applicable for this file

File: tests/csm/rest/test_query_deployment.py [hunk 2]
```diff
@@ -250,3 +250,53 @@ def test_45677(self):
...
+        Verify certificate details in GET system topology with default certificate.
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("Step 1: Send get certificate request")
+        result, err_msg = self.csm_obj.verify_certificate_details(expected_response=HTTPStatus.OK)
+        assert result, err_msg
+        self.log.info("##### Test ended -  %s #####", test_case_name)
+
+    #Test Not Ready
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_query_deployment.py:269]: Skip the test

File: libs/csm/rest/csm_rest_query_deployment.py [hunk 3]
```diff
@@ -207,3 +223,196 @@ def verify_storage_set(self, storage_set_id: str = None,
...
+                result = False
+        return result, err_msg
+
+    def verify_certificate_details(self, expected_response=HTTPStatus.OK):
+        """
+        Verify certificate details
+        """
+        get_response = self.get_certificate_topology()
+        result = get_response.status_code == expected_response
+        if result:
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_query_deployment.py:380]: add log here that "Status code check passed"

File: libs/csm/rest/csm_rest_query_deployment.py [hunk 4]
```diff
@@ -207,3 +223,196 @@ def verify_storage_set(self, storage_set_id: str = None,
...
+                self.log.info("Time details match successful")
+        return result, err_msg
+
+    def verify_serial_number(self, get_response_number):
+        """
+        Verify serial number for certificate
+        """
+        err_msg = ""
+        result = True
+        if result:
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_query_deployment.py:274]: Unnecessary if

File: libs/csm/rest/csm_rest_query_deployment.py [hunk 5]
```diff
@@ -207,3 +223,196 @@ def verify_storage_set(self, storage_set_id: str = None,
...
+    def verify_datetime_details(self, validity_str: str, get_response_date,
+                                get_response_time):
+        """
+        Verify what date certificate is valid before
+        """
+        err_msg = ""
+        output = []
+        flat_list = []
+        result = True
+        if result:
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_query_deployment.py:235]: Unnecessary if

File: libs/csm/rest/csm_rest_query_deployment.py [hunk 6]
```diff
@@ -207,3 +223,196 @@ def verify_storage_set(self, storage_set_id: str = None,
...
+                self.log.info("Serial number match successful")
+        return result, err_msg
+
+    def verify_version_number(self, get_response_version):
+        """
+        Verify certificate version
+        """
+        err_msg = ""
+        result = True
+        if result:
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_query_deployment.py:296]: Unnecessary if

File: libs/csm/rest/csm_rest_query_deployment.py [hunk 7]
```diff
@@ -207,3 +223,196 @@ def verify_storage_set(self, storage_set_id: str = None,
...
+
+    def verify_issuer_details(self, issuer_dict: dict):
+        """
+        Verify certificate issuer details
+        """
+        err_msg = ""
+        values_list = []
+        flat_list = []
+        result = True
+        if result:
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_query_deployment.py:324]: Unnecessary if

File: libs/csm/rest/csm_rest_query_deployment.py [hunk 8]
```diff
@@ -207,3 +223,196 @@ def verify_storage_set(self, storage_set_id: str = None,
...
+
+    def verify_subject_details(self, subject_dict: dict):
+        """
+        Verify certificate subject details
+        """
+        err_msg = ""
+        values_list = []
+        flat_list = []
+        result = True
+        if result:
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_query_deployment.py:353]: Unnecessary if

File: libs/csm/rest/csm_rest_query_deployment.py [hunk 9]
```diff
@@ -207,3 +223,193 @@ def verify_storage_set(self, storage_set_id: str = None,
...
+            result = False
+        else:
+            self.log.info("Date Verification successful")
+        self.log.info("Verify time in date details")
+        time_cmd = flat_list[2]
+        self.log.info("Time from get_response: %s", time_cmd)
+        self.log.info("Time from command: %s", time_cmd)
+        if time_cmd != get_response_time:
+            err_msg = err_msg + " Time details match failed"
+            self.log.error(err_msg)
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_query_deployment.py:262]: result should be set to False here

File: libs/csm/rest/csm_rest_query_deployment.py [hunk 10]
```diff
@@ -207,3 +223,193 @@ def verify_storage_set(self, storage_set_id: str = None,
...
+        resp = resp.decode() if isinstance(resp, bytes) else resp
+        self.log.info("Print resp: %s", resp)
+        for version_cmd in resp:
+            if "version" in version_cmd.lower():
+                version_cmd = "v" + version_cmd.strip().split()[1]
+                self.log.info("Printing version from command %s", version_cmd)
+                break
+        get_response_version = get_response_version.split(".")[1]
+        self.log.info("Printing get response version %s", get_response_version)
+        if version_cmd == get_response_version:
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_query_deployment.py:306]: result = version_cmd == get_response_version

File: libs/csm/rest/csm_rest_query_deployment.py [hunk 11]
```diff
@@ -207,3 +223,193 @@ def verify_storage_set(self, storage_set_id: str = None,
...
+            self.log.error(err_msg)
+            result = False
+        return result, err_msg
+
+    def verify_version_number(self, get_response_version: str):
+        """
+        Verify certificate version
+        """
+        err_msg = ""
+        result = True
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_query_deployment.py:294]: Check the comment below. Not required

File: libs/csm/rest/csm_rest_query_deployment.py [hunk 12]
```diff
@@ -207,3 +223,193 @@ def verify_storage_set(self, storage_set_id: str = None,
...
+        self.log.info("Print resp: %s", resp)
+        for elements in resp:
+            values_list.append(elements.strip().split("="))
+        for elements in values_list:
+            flat_list.extend(elements)
+        res_dct = {flat_list[i]: flat_list[i + 1] for i in range(0, len(flat_list), 2)}
+        del res_dct["issuer"]
+        self.log.info("res dict %s ", res_dct)
+        self.log.info("issuer dict %s ", issuer_dict)
+        if res_dct == issuer_dict:
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_query_deployment.py:334]: result = res_dct == issuer_dict

File: libs/csm/rest/csm_rest_query_deployment.py [hunk 13]
```diff
@@ -207,3 +223,193 @@ def verify_storage_set(self, storage_set_id: str = None,
...
+        self.log.info("Print resp: %s", resp)
+        for elements in resp:
+            values_list.append(elements.strip().split("="))
+        for elements in values_list:
+            flat_list.extend(elements)
+        res_dct = {flat_list[i]: flat_list[i + 1] for i in range(0, len(flat_list), 2)}
+        del res_dct["subject"]
+        self.log.info("res dict %s ", res_dct)
+        self.log.info("subject dict %s ", subject_dict)
+        if res_dct == subject_dict:
```
Comment (kachhwahadivya) [inline libs/csm/rest/csm_rest_query_deployment.py:362]: result = res_dct == subject_dict

File: libs/csm/rest/csm_rest_query_deployment.py (hunk 10 above)
Comment (aditipande96) [reply libs/csm/rest/csm_rest_query_deployment.py:306]: do you mean removing result = True from above and modifying this to if result = res_dct == issuer_dict; ?

**Review by ravindrachoudhari — APPROVED** [review]

**Review by kachhwahadivya — APPROVED** [review]

**Pull request merged**